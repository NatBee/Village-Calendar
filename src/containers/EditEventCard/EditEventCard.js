import React, { Component } from 'react';
import './EditEventCard.css';
import { connect } from 'react-redux';
import { deleteEventFromCalendar, editEventOnCalendar } from '../../helper/apiCall';

class EditEventCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  componentDidMount = () => {
    const title = this.props.event.title || '';
    const start = this.props.event.start || '';
    const end = this.props.event.end || '';
    const description = this.props.event.description || '';
    const location = this.props.event.location || '';
    this.setState({
      title, start, end, description, location
    })
  }

  deleteEvent = (e) => {
    deleteEventFromCalendar(this.props.calendarID, this.props.event.eventID)
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.name;
    const value = e.target.value;
    this.setState({ [field]: value });
  }

  editEvent = (e) => {
    editEventOnCalendar(this.props.calendarID, this.props.event.eventID, this.state);
  }


  render() {
    return(
      <div>
        <h1>Edit Event</h1>
        <form>
          <input type='text' name='title' value={this.state.title} onChange={this.handleChange} />
          <input type='text' name='start' value={this.state.start} onChange={this.handleChange} />
          <input type='text' name='end' value={this.state.end} onChange={this.handleChange} />
          <input type='text' name='description' value={this.state.description} onChange={this.handleChange} />
          <input type='text' name='location' value={this.state.location} onChange={this.handleChange} />
          <button onClick={this.editEvent}>Edit</button>
          <button onClick={this.deleteEvent}>Delete</button>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  event: store.event,
  calendarID: store.calendarID
})

export default connect(mapStateToProps)(EditEventCard);