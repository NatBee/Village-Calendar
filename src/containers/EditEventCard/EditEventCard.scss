.edit-card {
  width: 300px;
  display: inline-block;
  background-color: #A7B9AB;
  opacity: .85;
  padding: 50px 40px 36px 40px;
  margin: 60px;
}

.edit-card form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

h2 {
  font-size: 1.25em;
  margin: 0 0 20px;
}

.edit-card button { 
  background-color: #3E4D4E;
  height: 30px;
  color: #DBE3D4;
  font-family: 'Exo', sans-serif;
  border: 2px solid #293435;
  margin: 0 0 5px;
  width: 70px;
}

.edit-card button:hover {
  background-color: #293435;
}

.edit-card form input {
  margin: 10px 0 0px;
  font-size: 1em;
  padding: 2px 6px;
  width: 275px;
}

.edit-btns {
  display: flex;
}

.edit-card form .edit-cancel-btn {
  margin: 25px 40px 0 0px;
}

.edit-card form .edit-btn-btn {
  margin: 25px 0px 0 0px;
}

.edit-card form .edit-delete-btn {
  margin: 25px 0px 0 40px;
}
