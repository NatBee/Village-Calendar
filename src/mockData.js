export const mockData = {
  user: {
    user: 'Dk', 
    credential: 'Xg', 
    additionalUserInfo: 'Pg', 
    operationType: "signIn"
  },
  events: [
    {
      start: "2018-02-09T09:00:00-07:00",
      end: "2018-02-09T10:00:00-07:00",
      title: "Super Fun Party",
      description: "Fun with friends",
      eventID: "7js6e24fo0kin4qcotoegu37hg",
      location: 'Mi Casa'
    }
  ],
  token: 'token_number',
  calendarID: 'calendarID',
  village: [{natalie: 'nataliesmontoya@gmail.com'}],
  time: {
    startTime: "2018-02-28T19:00:00.000Z", 
    endTime: "2018-02-28T20:00:00.000Z"
  },
  event: {
    start: "2018-02-09T09:00:00-07:00",
    end: "2018-02-09T10:00:00-07:00",
    title: "Super Fun Party",
    description: "Fun with friends",
    eventID: "7js6e24fo0kin4qcotoegu37hg",
    location: 'Mi Casa'
  }
}