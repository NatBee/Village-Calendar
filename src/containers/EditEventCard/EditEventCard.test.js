/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { EditEventCard, mapStateToProps} from './EditEventCard';
import { mockData } from '../../mockData';

describe('EditEventCard', () => {
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<EditEventCard event={ mockData.event }/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockStore = {
      event: mockData.event,
      calendarID: mockData.calendarID,
      token: mockData.token
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  })

  it('should have a default state of event data', () => {
    const wrapper = shallow(<EditEventCard event={ mockData.event }/>);
    const expected = {
      title: mockData.event.title,
      start: mockData.event.start,
      end: mockData.event.end,
      description: mockData.event.description,
      location: mockData.event.location
    }
    expect(wrapper.state()).toEqual(expected);
  })

})