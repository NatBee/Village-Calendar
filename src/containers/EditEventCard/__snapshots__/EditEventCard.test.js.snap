// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEventCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditEventCard
    event={
      Object {
        "description": "Fun with friends",
        "end": "2018-02-09T10:00:00-07:00",
        "eventID": "7js6e24fo0kin4qcotoegu37hg",
        "location": "Mi Casa",
        "start": "2018-02-09T09:00:00-07:00",
        "title": "Super Fun Party",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Edit or Delete Event
        </h2>,
        <form>
          <input
            name="title"
            onChange={[Function]}
            placeholder="Event title"
            type="text"
            value="Super Fun Party"
          />
          <input
            name="start"
            onChange={[Function]}
            placeholder="Event start time"
            type="text"
            value="2018-02-09T09:00:00-07:00"
          />
          <input
            name="end"
            onChange={[Function]}
            placeholder="Event end time"
            type="text"
            value="2018-02-09T10:00:00-07:00"
          />
          <input
            name="description"
            onChange={[Function]}
            placeholder="Event description"
            type="text"
            value="Fun with friends"
          />
          <input
            name="location"
            onChange={[Function]}
            placeholder="Event location"
            type="text"
            value="Mi Casa"
          />
          <div
            className="edit-btns"
          >
            <button
              className="edit-cancel-btn"
              onClick={[Function]}
            >
              Cancel
            </button>
            <button
              className="edit-btn-btn"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="edit-delete-btn"
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </form>,
      ],
      "className": "edit-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit or Delete Event",
        },
        "ref": null,
        "rendered": "Edit or Delete Event",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="title"
              onChange={[Function]}
              placeholder="Event title"
              type="text"
              value="Super Fun Party"
            />,
            <input
              name="start"
              onChange={[Function]}
              placeholder="Event start time"
              type="text"
              value="2018-02-09T09:00:00-07:00"
            />,
            <input
              name="end"
              onChange={[Function]}
              placeholder="Event end time"
              type="text"
              value="2018-02-09T10:00:00-07:00"
            />,
            <input
              name="description"
              onChange={[Function]}
              placeholder="Event description"
              type="text"
              value="Fun with friends"
            />,
            <input
              name="location"
              onChange={[Function]}
              placeholder="Event location"
              type="text"
              value="Mi Casa"
            />,
            <div
              className="edit-btns"
            >
              <button
                className="edit-cancel-btn"
                onClick={[Function]}
              >
                Cancel
              </button>
              <button
                className="edit-btn-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="edit-delete-btn"
                onClick={[Function]}
              >
                Delete
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "title",
              "onChange": [Function],
              "placeholder": "Event title",
              "type": "text",
              "value": "Super Fun Party",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "start",
              "onChange": [Function],
              "placeholder": "Event start time",
              "type": "text",
              "value": "2018-02-09T09:00:00-07:00",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "end",
              "onChange": [Function],
              "placeholder": "Event end time",
              "type": "text",
              "value": "2018-02-09T10:00:00-07:00",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "description",
              "onChange": [Function],
              "placeholder": "Event description",
              "type": "text",
              "value": "Fun with friends",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "location",
              "onChange": [Function],
              "placeholder": "Event location",
              "type": "text",
              "value": "Mi Casa",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="edit-cancel-btn"
                  onClick={[Function]}
                >
                  Cancel
                </button>,
                <button
                  className="edit-btn-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>,
                <button
                  className="edit-delete-btn"
                  onClick={[Function]}
                >
                  Delete
                </button>,
              ],
              "className": "edit-btns",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "className": "edit-cancel-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "edit-btn-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "className": "edit-delete-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Edit or Delete Event
          </h2>,
          <form>
            <input
              name="title"
              onChange={[Function]}
              placeholder="Event title"
              type="text"
              value="Super Fun Party"
            />
            <input
              name="start"
              onChange={[Function]}
              placeholder="Event start time"
              type="text"
              value="2018-02-09T09:00:00-07:00"
            />
            <input
              name="end"
              onChange={[Function]}
              placeholder="Event end time"
              type="text"
              value="2018-02-09T10:00:00-07:00"
            />
            <input
              name="description"
              onChange={[Function]}
              placeholder="Event description"
              type="text"
              value="Fun with friends"
            />
            <input
              name="location"
              onChange={[Function]}
              placeholder="Event location"
              type="text"
              value="Mi Casa"
            />
            <div
              className="edit-btns"
            >
              <button
                className="edit-cancel-btn"
                onClick={[Function]}
              >
                Cancel
              </button>
              <button
                className="edit-btn-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="edit-delete-btn"
                onClick={[Function]}
              >
                Delete
              </button>
            </div>
          </form>,
        ],
        "className": "edit-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit or Delete Event",
          },
          "ref": null,
          "rendered": "Edit or Delete Event",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="title"
                onChange={[Function]}
                placeholder="Event title"
                type="text"
                value="Super Fun Party"
              />,
              <input
                name="start"
                onChange={[Function]}
                placeholder="Event start time"
                type="text"
                value="2018-02-09T09:00:00-07:00"
              />,
              <input
                name="end"
                onChange={[Function]}
                placeholder="Event end time"
                type="text"
                value="2018-02-09T10:00:00-07:00"
              />,
              <input
                name="description"
                onChange={[Function]}
                placeholder="Event description"
                type="text"
                value="Fun with friends"
              />,
              <input
                name="location"
                onChange={[Function]}
                placeholder="Event location"
                type="text"
                value="Mi Casa"
              />,
              <div
                className="edit-btns"
              >
                <button
                  className="edit-cancel-btn"
                  onClick={[Function]}
                >
                  Cancel
                </button>
                <button
                  className="edit-btn-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="edit-delete-btn"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "title",
                "onChange": [Function],
                "placeholder": "Event title",
                "type": "text",
                "value": "Super Fun Party",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "start",
                "onChange": [Function],
                "placeholder": "Event start time",
                "type": "text",
                "value": "2018-02-09T09:00:00-07:00",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "end",
                "onChange": [Function],
                "placeholder": "Event end time",
                "type": "text",
                "value": "2018-02-09T10:00:00-07:00",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "description",
                "onChange": [Function],
                "placeholder": "Event description",
                "type": "text",
                "value": "Fun with friends",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "location",
                "onChange": [Function],
                "placeholder": "Event location",
                "type": "text",
                "value": "Mi Casa",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="edit-cancel-btn"
                    onClick={[Function]}
                  >
                    Cancel
                  </button>,
                  <button
                    className="edit-btn-btn"
                    onClick={[Function]}
                  >
                    Edit
                  </button>,
                  <button
                    className="edit-delete-btn"
                    onClick={[Function]}
                  >
                    Delete
                  </button>,
                ],
                "className": "edit-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cancel",
                    "className": "edit-cancel-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-btn-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "className": "edit-delete-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
